src/any_llm/providers/huggingface/huggingface.py

```
@@ -49,6 +49,23 @@
     from any_llm.types.responses import ParsedResponse, Response, ResponsesParams
 
 
+def _normalize_tool_calls(tool_calls: Any) -> Any:
```

Ruff (0.15.17)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `tool_calls`

(ANN401)